cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/client/AzureClient.java

```diff
@@ -863,4 +865,16 @@ private void handleAuthException(Runnable function) {
     public Completable deleteGenericResourceByIdAsync(String databaseServerId) {
         return handleAuthException(() -> azure.genericResources().deleteByIdAsync(databaseServerId));
     }
+
+    public void deleteDiskEncryptionSet(String resourceGroup, String diskEncryptionSetName) {
+        // The Disk encryption set operations are not exposed in public API, so have to rely on underlying DiskEncryptionSetInner
+        DiskEncryptionSetsInner dSetsIn = azureClientCredentials.getComputeManager().inner().diskEncryptionSets();
+        dSetsIn.delete(resourceGroup, diskEncryptionSetName);
```

Comment:
what happens if the encryption set is deleted by the customer

Reply:
If environment exists and DES is deleted by customer, Datahub(as of now only datahub uses des) create should result in failure. cc: @lajosrodek 
During deletion of environment, DES deletion will not happen from our side as `AzureClient.getDiskEncryptionSet` will return `null`. However, as we have DES stored in `environment_parameters` - respective cloudResource will be deleted.

Reply:
- what if customer deletes the DES permission and we try to delete
- are you sure we get a null ? How we check that the delete is success ?

Reply:
`Microsoft.Compute/diskEncryptionSets/delete` is required to delete DES. If this is not present we would not be able to delete.
yes, `null` is returned when DES doesn't exists in cloud. To check that DES is indeed deleted - wait for sometime(as there is latency to get correct status) and recheck if `AzureClient.getDiskEncryptionSet` returns `null`.

Reply:
but we need to check the deletion status right ?

Reply:
> what happens if the encryption set is deleted by the customer
Subsequent DH launches will surely fail in that case. 😄 This is indeed a possibility as long as the DES is not associated with any disk resource. The handling of that check is planned in [CB-11789](https://jira.cloudera.com/browse/CB-11789); that shall provide sufficient safeguards to block env deletion if the DES is in use by experiences. (The exact logic needs some elaboration, however.)

Reply:
The cloud API call to delete the DES doesn’t return anything. However, it may result in
```
@throws IllegalArgumentException thrown if parameters fail the validation      
@throws CloudException thrown if the request is rejected by server
@throws RuntimeException all other wrapped checked exceptions if the request fails to be sent
```
As this is a blocking call, we need not to check the status of deletion - it either succeeds or results in error.